src/app/loadout-drawer/loadout-item-conversion.ts

```diff
@@ -44,15 +45,44 @@ export function getItemsFromLoadoutItems(
       const modsForBucket =
         loadoutItem.equip && modsByBucket ? modsByBucket[item.bucket.hash] ?? [] : [];
 
-      let overrides = loadoutItem.socketOverrides;
+      let overrides = loadoutItem.socketOverrides || {};
 
+      // Apply armour mods to the socket overrides
       for (const modHash of modsForBucket) {
         const socket = item.sockets?.allSockets.find((s) => plugFitsIntoSocket(s, modHash));
         if (socket) {
           overrides = { ...overrides, [socket?.socketIndex]: modHash };
         }
       }
 
+      // Hydrate the empty sockets for fragments as overrides. This ensures that the stats that
+      // result from the overridden item are correct.
+      if (
+        item.bucket.hash === BucketHashes.Subclass &&
+        item.sockets?.allSockets &&
+        defs.isDestiny2()
+      ) {
+        const aspects = getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.Aspects);
+        let activeFragments = 0;
```

Comment:
If I'm reading this code right, this isn't active fragments, this is fragment capacity, right? This bit is just trying to count up how many fragment slots there are?

Reply:
Yeah you are correct, I find the capacity naming weird also though (I realise active isn't that great either). Basically the fragment capacity is always 6, the summed `aspect.plugged?.plugDef.plug.energyCapacity?.capacityValue` will tell us how many of those 6 will be active fragments, meaning that they will contribute to stats and have active perks.